pkg/state/impl/inmem/collection.go

```diff
@@ -440,23 +441,39 @@ func (collection *ResourceCollection) WatchAll(ctx context.Context, ch chan<- st
 
 	go func() {
 		// send initial contents if they were captured
-		for _, res := range bootstrapList {
-			if !channel.SendWithContext(ctx, ch,
-				state.Event{
-					Type:     state.Created,
-					Resource: res,
-				},
-			) {
-				return
-			}
-		}
+		if options.BootstrapContents {
+			switch {
+			case singleCh != nil:
+				for _, res := range bootstrapList {
+					if !channel.SendWithContext(ctx, singleCh,
+						state.Event{
+							Type:     state.Created,
+							Resource: res,
+						},
+					) {
+						return
+					}
+				}
 
-		bootstrapList = nil
+				if !channel.SendWithContext(ctx, singleCh, state.Event{Type: state.Bootstrapped}) {
+					return
+				}
+			case aggCh != nil:
+				events := slices.Map(bootstrapList, func(r resource.Resource) state.Event {
+					return state.Event{
+						Type:     state.Created,
+						Resource: r,
+					}
+				})
 
-		if options.BootstrapContents {
-			if !channel.SendWithContext(ctx, ch, state.Event{Type: state.Bootstrapped}) {
-				return
+				events = append(events, state.Event{Type: state.Bootstrapped})
+
+				if !channel.SendWithContext(ctx, aggCh, events) {
+					return
+				}
 			}
+
+			bootstrapList = nil
```

Comment:
No more reads or writes after than, any reason for nilling it?

Reply:
IIRC runtime/compiler knows that variable is no longer used if the function is big enough. Since `WatchAll` returns it means that only this goroutine could probably keep this variable.

Anyway - I don't mind keep that block, but maybe we should comment the reason for doing this?

Reply:
yep, added